### DISCUSSION TITLE
Why is my excludeFiles glob not working for serverless functions?

### DISCUSSION THREAD

**Discussion by meds** [vercel/community#3348] [Help]
### Summary

I've hit the 50megabyte limit in vercel functions, after investigating a bit I found a lot of packages were being included that are not used server side. As an experiment I added a vercel.json at the root folder with this content:

```{
    "functions": {
        "pages/api/**/*": {
            "excludeFiles": "node_modules/.pnpm/@mui+material@5.14.1_@emotion+react@11.11.1_@emotion+styled@11.11.0_@types+react@18.2.15_react-dom@18.2.0_react@18.2.0/**/*|node_modules/.pnpm/react-dom@18.2.0_react@18.2.0/**/*|public/**/*"
        }
    }
}
```

have also tried using outputFileTracingExlcudes (and variations):

```

module.exports = {
  experimental: {
    appDir: true,
    outputFileTracingExcludes: {
      'api/send-prescription-reminder': ['./node_modules/.pnpm/@mui+material@5.14.1_@emotion+react@11.11.1_@emotion+styled@11.11.0_@types+react@18.2.15_react-dom@18.2.0_react@18.2.0/**/*'],
      '/api/send-prescription-reminder': ['./node_modules/**/*', './node_modules/.pnpm/@mui+material@5.14.1_@emotion+react@11.11.1_@emotion+styled@11.11.0_@types+react@18.2.15_react-dom@18.2.0_react@18.2.0/**/*'],
      '/api/**/*': ['node_modules/.pnpm/@mui+material@5.14.1_@emotion+react@11.11.1_@emotion+styled@11.11.0_@types+react@18.2.15_react-dom@18.2.0_react@18.2.0/**/*']
    }
  },
  typescript: {
    ignoreBuildErrors: false,
  },
  compiler: {
    emotion: true,
  },
  webpack(config) {
    config.module.rules.push({
      test: /\.svg$/,
      issuer: {
        and: [/\.(js|ts)x?$/],
      },
      use: ['@svgr/webpack'],
    });
    return config;
  },
}
```

Unfortunately it made no different to the build size and I could see it was still factoring in the excluded files above in the packages break down.

Any ideas why they're being ignored?

### Steps to Reproduce

Add a package and use pnpm to install

Exclude it from serverless functions using vercel.json

Observe: It's still being included in the final bundle

**Comment by amyegan** [comment]
Hey @meds. What framework does your project use? 

Some frameworks have their own bundling strategies and need slightly different configuration. For example, Next.js could use `outputFileTracingExcludes` in `next.config.js` to specify files to exclude in output file tracing.
